examples/test_example.rs

```diff
@@ -0,0 +1,39 @@
+fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
What does this example showcase?

Reply:
This is only temporary, will be removed once this PR is ready. I use this for easier debugging

Reply:
Example has been removed.